### ISSUE TITLE
v2.2.2-rc8: Error during creating an instance

### ISSUE THREAD

**Issue by ronie-z-wave** [maros/Zway-Presence#1]
BaseModule is installed and running.

Think this has to do with the time form field. We've added the correct library for this so that is now working. Unfortunately it has the format 00:00:00 am/pm by default. So maybe you need to change some stuff.
For example this is already done in the apps LogicalRules or ScheduledScene.

```
"schema": {
      "properties": {
            "time": {
                   "format":"time",
                   "type": "string",
               "required": true
            }
      }
},
"options": {
    "fields": {
         "time": {
                    "type": "time",
                    "maskString": "99:99",
                    "dateFormat": "HH:mm"
         }
     }
}
```

should work ... but strange that it doesn't in your module. Could you please have a look?

log:

```
 [2016-04-07 09:46:46.818] [I] [core] Instantiating module 10 from class BaseModule
[2016-04-07 09:46:46.824] [I] [core] --- Starting module Base Module
[2016-04-07 09:46:46.841] [I] [core] [BaseModule-10] Init callbacks
[2016-04-07 09:47:21.196] [I] [core] Notification: error (core): Cannot instantiate module: Presence: TypeError: undefined is not a function
[2016-04-07 09:47:21.230] [I] [core] TypeError: undefined is not a function
    at AutomationController.instantiateModule (automation/classes/AutomationController.js:394:24)
    at AutomationController.createInstance (automation/classes/AutomationController.js:833:23)
    at ZAutomationAPIWebRequest._.extend.createInstance (automation/ZAutomationAPIProvider.js:1027:40)
    at ZAutomationAPIWebRequest.ZAutomationWebRequest.handleRequest (automation/WebserverRequestRouter.js:258:41)
    at automation/WebserverRequestRouter.js:32:35
    at WebServer.document_root (automation/Webserver.js:29:11)
```

**Comment by maros** [comment]
The time format is already fixed in master ( https://github.com/maros/Zway-Presence/commit/388a8f38efc3a4a2abeb284144f9e98135e9e043 ), and parseTime in BaseModule master now accepts both formats. Will release both modules soon since 2.2.2 seems to be round the corner. 

However, I'm a bit confused by the stack trace. I don't even know how to produce such an error message except by calling "undefined()" directly - eg. "var x = undefined; x();" produces a different message. This very much reminds me of the error message I saw in https://github.com/Z-Wave-Me/home-automation/issues/338. Could it be that there was a sublte change in how stacktraces/errors are produced? Additionally I don't see my code in the stacktrace?

**Comment by ronie-z-wave** [comment]
Ok nice :)

Indeed in both cases something is undefined ... but for different reasons. Hmm I suspect there is a function call like x.function('foobar') but x has no function with name 'function' ...

I'll check this. Maybe the app is not root of the problem ;)

**Comment by ronie-z-wave** [comment]
FYI:
Interesting now I'm getting a more informative error:

```
[2016-04-07 14:23:42.051] [I] [core] Instantiating module 15 from class Presence
[2016-04-07 14:23:42.058] [I] [core] --- Starting module Presence state
[2016-04-07 14:23:42.091] [I] [core] Creating device switchBinary Presence_presence_15
[2016-04-07 14:23:42.150] [I] [core] Creating device switchBinary Presence_vacation_15
[2016-04-07 14:23:42.178] [I] [core] Creating device switchBinary Presence_night_15
[2016-04-07 14:23:42.214] [I] [core] Notification: error (core): Cannot instantiate module: Presence: TypeError: Cannot read property 'getMinutes' of undefined
[2016-04-07 14:23:42.259] [I] [core] TypeError: Cannot read property 'getMinutes' of undefined
    at automation/userModules/Presence/index.js:47:33
    at Function._.each._.forEach (automation/lib/underscore.js:146:17)
    at Presence.init (automation/userModules/Presence/index.js:42:7)
    at AutomationController.instantiateModule (automation/classes/AutomationController.js:415:22)
    at AutomationController.createInstance (automation/classes/AutomationController.js:833:23)
    at ZAutomationAPIWebRequest._.extend.createInstance (automation/ZAutomationAPIProvider.js:1027:40)
    at ZAutomationAPIWebRequest.ZAutomationWebRequest.handleRequest (automation/WebserverRequestRouter.js:258:41)
    at automation/WebserverRequestRouter.js:32:35
    at WebServer.document_root (automation/Webserver.js:29:11)
```

but if it breaks here `instance = new global[module.meta.id](instanceModel.id, self);` something is wrong with the index.js ...

But you've already fixed that, so don't worry about that

**Comment by maros** [comment]
Fixed in 1.10
